packages/core/src/sync-store/index.ts

```diff
@@ -305,22 +372,22 @@ export const createSyncStore = ({
       }
 
       const result = new Map<
-        Filter,
+        Filter | Factory,
         { fragment: Fragment; intervals: Interval[] }[]
       >();
 
       // NOTE: `interval[1]` must be rounded down in order to offset the previous
       // rounding.
 
-      for (let i = 0; i < filters.length; i++) {
-        const filter = filters[i]!;
+      index = 0;
+
+      for (const filter of filters) {
         const fragments = getFragments(filter);
         result.set(filter, []);
-        for (let j = 0; j < fragments.length; j++) {
-          const fragment = fragments[j]!;
+
+        for (const fragment of fragments) {
           const intervals = rows
-            .filter((row) => row.filter === `${i}`)
-            .filter((row) => row.fragment === `${j}`)
+            .filter((row) => row.fragment === `${index}`)
```

Comment:
In the query, it seems like we increment the index on the first pass (such that the first value will be `'1'`, but here we are incrementing after the first pass.